File: webapp/models/table.py
import sqlalchemy as sa
import typing as t
from . import db
from sqlalchemy import and_, cast, or_, orm
if t.TYPE_CHECKING:
    from datetime import datetime




class Table(db.Model):
    """
    Represents a single table in a restaurant and the number of people it can fit
    """
    table_id = sa.Column(sa.Integer, primary_key=True)
    restaurant_id = sa.Column(sa.Integer, sa.ForeignKey('restaurant.restaurant_id'), nullable=False)
    reservation_id = sa.Column(sa.Integer, sa.ForeignKey('reservation.reservation_id'))
    table_number = sa.Column(sa.Integer, nullable=False)
    capacity = sa.Column(sa.Integer, nullable=False)
    location_description = sa.Column(sa.String(200))  # Description of the table's location
    is_reserved:bool = sa.Column(sa.Boolean, default=False)
    customer_id:int = sa.Column(sa.Integer, sa.ForeignKey('user.user_id'))
    created_at:'datetime' = sa.Column(sa.Date)
    updated_at:'datetime' = sa.Column(sa.Date)
    deleted_at:'datetime' = sa.Column(sa.DateTime, nullable=True)


    customer = orm.relationship('User', foreign_keys='User.user_id')
    reservation = orm.relationship('Reservation', foreign_keys='Reservation.reservation_id')

File: internal/models/table.go
package models

import (
	"gorm.io/gorm"
	// "time"
)

type Table struct {
	gorm.Model
	TableID             uint   `gorm:"primaryKey;autoIncrement"`
	RestaurantID        uint   `gorm:"not null"`
	ReservationID       *uint  // It's a pointer to allow nil value when no reservation is associated
	TableNumber         uint   `gorm:"not null"`
	Capacity            uint   `gorm:"not null"`
	LocationDescription string `gorm:"size:200"` // Description of the table's location
	IsReserved          bool   `gorm:"default:false"`
	CustomerID          *uint  // It's a pointer to allow nil value when no customer is associated

	// Define relationships
	// Restaurant   Restaurant `gorm:"foreignKey:RestaurantID"`
	// Reservation  Reservation `gorm:"foreignKey:ReservationID"`
	// Customer     User        `gorm:"foreignKey:CustomerID"`
}

// Override the default table name.
func (Table) TableName() string {
	return "tables"
}